extractor/src/main/java/org/schabi/newpipe/extractor/utils/LocaleCompat.java

```diff
@@ -0,0 +1,43 @@
+package org.schabi.newpipe.extractor.utils;
+
+import java.util.Locale;
+
+/**
+ * This class contains a simple implementation of {@link Locale#forLanguageTag(String)} for Android
+ * API levels below 21 (Lollipop). This is needed as core library desugaring does not backport that
+ * method as of this writing.
+ *
+ * Relevant issue: https://issuetracker.google.com/issues/171182330
+ */
+public final class LocaleCompat {
+    private LocaleCompat() {
+    }
+
+    // Source: The AndroidX LocaleListCompat class's private forLanguageTagCompat() method.
+    // Use Locale.forLanguageTag() on Android API level >= 21 / Java instead.
+    public static Locale forLanguageTag(final String str) {
+        if (str.contains("-")) {
+            String[] args = str.split("-", -1);
+            if (args.length > 2) {
+                return new Locale(args[0], args[1], args[2]);
+            } else if (args.length > 1) {
+                return new Locale(args[0], args[1]);
+            } else if (args.length == 1) {
+                return new Locale(args[0]);
+            }
+        } else if (str.contains("_")) {
+            String[] args = str.split("_", -1);
+            if (args.length > 2) {
+                return new Locale(args[0], args[1], args[2]);
+            } else if (args.length > 1) {
+                return new Locale(args[0], args[1]);
+            } else if (args.length == 1) {
+                return new Locale(args[0]);
+            }
+        } else {
+            return new Locale(str);
+        }
+
+        throw new IllegalArgumentException("Can not parse language tag: [" + str + "]");
```

Comment:
Note: this is unreachable since `args.length` > 0

Reply:
Yeah, I copied over the method implementation from AndroidX's [`LocaleListCompat`](https://developer.android.com/reference/androidx/core/os/LocaleListCompat) class, as noted in the comment above the method.